### PR TITLE
LPS-116949 Membership - management toolbar loses full width in Add modal

### PR DESCRIPTION
**Pull request by markocikos**
This is a bugfix, see full description and steps to reproduce in https://issues.liferay.com/browse/LPS-116949

The root cause of the issue is the `dialog-iframe-popup` class not setting on SPA navigation in modal iframe. We had this issue in `openModal`, fixed it with [this navigation listener](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/modal/Modal.js#L426-L435). In this PR, we are porting the fix to legacy modules.

I noticed that the `dialog-iframe-popup` is also not setting on the first opening of the modal. This is because we are not setting `p_p_id` in url, likely because of friendly-url-routes. No `p_p_id` leads to no `namespace` in modal and we cannot correctly append `bodyCssClass` param to iframe url. To fix this, in this PR we are adding `p_p_id` to url.

---

I noticed one more visual bug that I didn't know the best way to fix. @eduardoallegrini maybe you can help with this one.

A scrollbar appears on the right of the modal, sometimes immediately and sometimes on hover on header:
screen

It seems to be caused by [this overflow](https://github.com/liferay/liferay-portal/blob/master/modules/util/sass-compiler-jni/src/test/resources/com/liferay/sass/compiler/jni/internal/dependencies/clay-css/bootstrap/_modal.scss#L13), that we probably could remove or override. But, this is no simple scss file.. I don't know how it builds and integrates with Clay and portal.

### CODE REVIEW

**Comment by jbalsas** [comment]
Hey @markocikos, @dgarciasarai just reported [Flickering styles in modals](https://issues.liferay.com/browse/LPS-117488). I'm assuming that's the case where we can't get all the info from the URL?

File: modules/apps/frontend-js/frontend-js-aui-web/src/main/resources/META-INF/resources/liferay/util.js [hunk 1]
```diff
@@ -1108,6 +1108,25 @@
 
 			event.win.focus();
 
+			var iframeWindow = event.win;
+
+			if (iframeWindow.Liferay.SPA) {
+				var beforeScreenFlipHandler = iframeWindow.Liferay.on(
```
[the remark this answers is not in the document]
Comment (markocikos) [reply modules/apps/frontend-js/frontend-js-aui-web/src/main/resources/META-INF/resources/liferay/util.js:1114]: In my understanding, using `once` means listener is detached after first trigger. But here, we want to preserve it over multiple triggers. User can navigate multiple times before closing the modal.

**Comment by markocikos** [comment]
> Hey @markocikos, @dgarciasarai just reported [Flickering styles in modals](https://issues.liferay.com/browse/LPS-117488). I'm assuming that's the case where we can't get all the info from the URL?

Yep, that looks like another case of missing `p_p_id` in param.

File: modules/apps/frontend-js/frontend-js-aui-web/src/main/resources/META-INF/resources/liferay/util.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (jbalsas) [reply modules/apps/frontend-js/frontend-js-aui-web/src/main/resources/META-INF/resources/liferay/util.js:1114]: Got it! I misread it! 👍

[the remark this answers is not in the document]
Comment (markocikos) [reply modules/apps/frontend-js/frontend-js-aui-web/src/main/resources/META-INF/resources/liferay/util.js:1114]: Yep, it triggers only once.

**Comment by jbalsas** [comment]
ci:test:relevant

**Comment by markocikos** [comment]
Rebased and force-pushed, to try to get rid of unrelated CI failure. No changes.

**Comment by markocikos** [comment]
ci:test:sf

**Comment by markocikos** [comment]
ci:test:relevant

**Comment by markocikos** [comment]
ci:test:relevant

**Comment by markocikos** [comment]
With tests in green and comments looking resolved, I'm going to push this.

**Comment by markocikos** [comment]
ci:forward

**Pull request closed without merging**